File: java-samples/shinny-tokens/contracts/src/main/java/com/r3/developers/csdetemplate/utxoexample/states/GoldState.java
package com.r3.developers.csdetemplate.utxoexample.states;

import com.r3.developers.csdetemplate.utxoexample.contracts.GoldContract;
import net.corda.v5.crypto.SecureHash;
import net.corda.v5.ledger.utxo.BelongsToContract;
import net.corda.v5.ledger.utxo.ContractState;

import java.math.BigDecimal;
import java.security.PublicKey;
import java.util.List;

@BelongsToContract(GoldContract.class)
public class GoldState implements ContractState {

    private SecureHash issuer;
    private String symbol;
    private BigDecimal value;
    private SecureHash owner;
    public List<PublicKey> participants;

    public GoldState(SecureHash issuer, String symbol, BigDecimal value, List<PublicKey> participants, SecureHash owner) {
        this.issuer = issuer;
        this.symbol = symbol;
        this.value = value;
        this.participants = participants;
        this.owner = owner;
    }

    public SecureHash getIssuer() {
        return issuer;
    }

    public String getSymbol() {
        return symbol;
    }

    public BigDecimal getValue() {
        return value;
    }

    public SecureHash getOwner() {
        return owner;
    }

    @Override
    public List<PublicKey> getParticipants() {
        return participants;
    }





}

File: kotlin-samples/shinny-tokens/contracts/src/main/kotlin/com/r3/developers/samples/tokens/states/GoldState.kt
package com.r3.developers.samples.tokens.states

import com.r3.developers.samples.tokens.contracts.GoldContract
import net.corda.v5.application.crypto.DigestService
import net.corda.v5.crypto.SecureHash
import net.corda.v5.ledger.utxo.BelongsToContract
import net.corda.v5.ledger.utxo.ContractState
import net.corda.v5.ledger.utxo.observer.UtxoLedgerTokenStateObserver
import net.corda.v5.ledger.utxo.observer.UtxoToken
import net.corda.v5.ledger.utxo.observer.UtxoTokenFilterFields
import net.corda.v5.ledger.utxo.observer.UtxoTokenPoolKey
import java.math.BigDecimal
import java.security.PublicKey

@BelongsToContract(GoldContract::class)
data class GoldState(
    val issuer: SecureHash,
    val owner: SecureHash,
    val symbol: String,
    val amount: BigDecimal,
    private val participants: List<PublicKey>) : ContractState {
    override fun getParticipants(): List<PublicKey> {
        return  participants
    }

}

/*
By implementing the UtxoLedgerTokenStateObserver, this observer will generate fungible states/tokens for
each produced gold state when persisting a finalized transaction to the vault.
 */
class GoldStateObserver: UtxoLedgerTokenStateObserver<GoldState>{
    override fun getStateType(): Class<GoldState> {
        return GoldState::class.java
    }

    override fun onCommit(state: GoldState, digestService: DigestService): UtxoToken {
        //generate a pool with key - type, issuer and symbol to mint the tokens
        val poolKey = UtxoTokenPoolKey(GoldState::class.java.getName(), state.issuer, state.symbol)
        return UtxoToken(poolKey, state.amount, UtxoTokenFilterFields(null, state.owner))
    }

}